.github/workflows/playwright.yml

```
@@ -1,19 +1,196 @@
+      MM_ADMIN_EMAIL: sysadmin@sample.mattermost.com
+      MM_ADMIN_USERNAME: sysadmin
+      MM_ADMIN_PASSWORD: Sys@dmin-sample1
+      TEST_DATABASE_URL: postgres://mmuser:mostest@localhost:5432/mattermost_test
+      MM_SERVICESETTINGS_ENABLEDEVELOPER: true
     steps:
       - name: ci/checkout-repo
         uses: actions/checkout@ac593985615ec2ede58e132d2e21d2b1cbd6127c # v3.3.0
+
+      - name: ci/setup-go
+        uses: actions/setup-go@6edd4406fa81c3da01a34fa6f6343087c207a568 # v3.5.0
+        with:
+          go-version: "${{ env.GO_VERSION }}"
+          cache: true
+
+      - name: ci/setup-node
+        uses: actions/setup-node@64ed1c7eab4cce3362f8c340dee64e5eaeef8f7c # v3.6.0
+        with:
+          node-version-file: ".nvmrc"
+          # cache: "npm"
+          # cache-dependency-path: webapp/package-lock.json
+
+      - name: ci/create-admin-user
+        run: |
+          STATUSCODE=$(curl -X POST -H "Content-Type: application/json" -d '{"email": "'${MM_ADMIN_EMAIL}'", "username": "'${MM_ADMIN_USERNAME}'", "password": "'${MM_ADMIN_PASSWORD}'"}' ${MM_SERVICESETTINGS_SITEURL}/api/v4/users -w "%{http_code}" -o /dev/stderr)
+          if test $STATUSCODE -ne 201; then exit 1; fi
+
+      - name: ci/install-go-dependencies
+        run: go mod tidy
+
+      - name: ci/install-plugin-webapp-deps
+        run: cd webapp && npm ci
+
+      - name: ci/build-and-install-plugin
+        run: make deploy
+        env:
+          PLUGIN_E2E_MOCK_OAUTH_SERVER_URL: http://172.17.0.1:8080
+          E2E_TESTING: true
+
+      # - name: ci/checkout-mattermost-monorepo
+      #   uses: actions/checkout@ac593985615ec2ede58e132d2e21d2b1cbd6127c # v3.3.0
+      #   with:
+      #     path: ../mattermost
+      #     repository: mattermost/mattermost
+
+      - name: ci/checkout-mattermost-monorepo
+        run: |
+          git clone https://github.com/mattermost/mattermost.git ../mattermost
+
+      - name: ci/setup-node-for-mattermost-monorepo
+        uses: actions/setup-node@64ed1c7eab4cce3362f8c340dee64e5eaeef8f7c # v3.6.0
+        with:
+          node-version-file: "../mattermost/.nvmrc"
+
+      - name: ci/install-mattermost-monorepo-webapp-deps
+        run: |
+          cd ../mattermost/webapp
+          npm i
+
+      - name: ci/install-mattermost-monorepo-playwright-deps
+        run: |
+          cd ../mattermost/e2e-tests/playwright
+          npm i
+          npx playwright install --with-deps
+
+      - name: ci/setup-node-for-mattermost-plugin
+        uses: actions/setup-node@64ed1c7eab4cce3362f8c340dee64e5eaeef8f7c # v3.6.0
+        with:
+          node-version-file: ".nvmrc"
+
+      - name: ci/install-plugin-playwright-deps
+        run: |
+          cd e2e/playwright
+          npm ci
+
+      - name: ci/tsc
+        run: |
+          cd e2e/playwright
+          npm run tsc
+
+      # - name: ci/lint
+      #   run: |
+      #     cd e2e/playwright
+      #     npm run lint
```

Since all of the other steps are descriptive:

```suggestion
      - name: ci/tsc-plugin-playwright
        run: |
          cd e2e/playwright
          npm run tsc

      # - name: ci/lint-plugin-playwright
      #   run: |
      #     cd e2e/playwright
      #     npm run lint
```